etc/sync-function-template.js

```diff
@@ -8,14 +8,14 @@ function(doc, oldDoc) {
 
   // Check that a given value is a valid ISO 8601 format date string with optional time and time zone components
   function isIso8601DateTimeString(value) {
-    var regex = new RegExp('^(([\\+-]?[0-9]{4})-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01]))([T ]([01][0-9]|2[0-4])(:[0-5][0-9])?(:[0-5][0-9])?([\\.,][0-9]{1,3})?)?([zZ]|([\\+-])([01][0-9]|2[0-3]):?([0-5][0-9])?)$');
+    var regex = new RegExp('^(([0-9]{4})-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01]))([T ]([01][0-9]|2[0-4])(:[0-5][0-9])?(:[0-5][0-9])?([\\.,][0-9]{1,3})?)?([zZ]|([\\+-])([01][0-9]|2[0-3]):?([0-5][0-9])?)?$');
 
     return regex.test(value);
   }
 
   // Check that a given value is a valid ISO 8601 date string without time and time zone components
   function isIso8601DateString(value) {
-    var regex = new RegExp('^(([\\+-]?[0-9]{4})-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01]))$');
+    var regex = new RegExp('^(([0-9]{4})-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01]))$');
```

Comment:
The positive/negative sign prefix for 4-digit years is not technically included in the [simplified](http://www.ecma-international.org/ecma-262/5.1/#sec-15.9.1.15) version of ISO 8601 supported by ECMAScript. While the specification does specify support for 6-digit years that **must** be prefixed by positive or negative, I have elected to remove the optional unary operator for the time being rather than add support for [extended years](http://www.ecma-international.org/ecma-262/5.1/#sec-15.9.1.15.1). This simplification makes it easier to compare dates without time and time zone components, but it is a potential limitation that may be worth revisiting later.